File: src/main/java/org/library/Borrowers.java
package org.library;

import java.util.HashMap;

public class Borrowers {
    private HashMap<String, Borrower> borrowers;

    public Borrowers(){
        borrowers = new HashMap<>();
    }

    public void addBorrower(Borrower b){
        borrowers.put(b.getUsername(), b);
    }

    public Borrower getBorrower(String username) {
        return borrowers.get(username);
    }

    public int getBorrowersSize() {return borrowers.size();}
}

File: main/Borrowers.js
export default class Borrowers {
  constructor() {
    this.borrowers = new Map();
  }

  addBorrower(b) {
    const username =
      typeof b.getUsername === "function" ? b.getUsername() : b.username;
    this.borrowers.set(username, b);
  }

  getBorrower(username) {
    return this.borrowers.get(username) || null;
  }

  getBorrowersSize() {
    return this.borrowers.size;
  }
}
